### PR TITLE
[PRO-289] Load resources from ProPro API and add search/filtering

### PR DESCRIPTION
**Pull request by tasandberg**
Changes
---
* Get rid of route loader and use react-query to load resources from API.
* Adds Resource and ResourceTags TS types.
* Adds ResourceTag translations
* Adds new filtering functionality:
  * Search
  * Tags
  * Location
* All new filter UI is fully translated (I think??)

Testing
---
* Generally click around the resources page in preview, trying all features and checking functionality of searching, tag, and location filters.
* Proper list rendering.
* Proper resource display.

image

This branch completes PRO-287

### CODE REVIEW

**Review by eddie-m-m — COMMENTED** [review]
Everything appears to work. 

The search results are a bit jumpy to the eye. 
Has an `AnimatedList`wrapper like our other lists been considered?

**Comment by tasandberg** [comment]
@eddie-m-m It was a separate issue, but once I fixed the flickering, I added animated list as well.

File: public/locales/en/translation.json [hunk 1]
```diff
@@ -251,7 +251,29 @@
...
     "suggestResource": "Suggest a resource",
-    "title": "Resources"
+    "title": "Resources",
+    "searchPlaceholder": "Search for a resource",
+    "locationFilter": {
+      "title": "Location",
+      "placeholder": "City, State or Zip code",
+      "within": "Within",
+      "of": "of",
+      "miles": "miles"
```
Comment (tasandberg) [inline public/locales/en/translation.json:261]: Need to add these to the es translations

**Comment by yuuks** [comment]
Was testing it out today and noticed a couple things.

- when entering a range with zipcode, and then searching, the results seem to ignore the filter
Screenshot 2024-03-22 at 11 43 34 AM

- The tag pills on the resource card appear clickable when i hover over them, but don't do anything when clicked. (this isn't really a big deal, just a small detail I noticed)

Other than that seems to function well, very cool! 👍

**Comment by tasandberg** [comment]
@yuuks Such a good catch-- looks like search is clearing all filters. Thanks!

**Comment by yuuks** [comment]
I was just testing on the preview this morning and still seeing the issue with the zipcode
Screenshot 2024-03-23 at 11 01 09 AM

**Comment by tasandberg** [comment]
> I was just testing on the preview this morning and still seeing the issue with the zipcode

Wasn't able to reproduce this locally or in the preview. I'm wondering if your browser cache still has the old code. Can you try hard refreshing (cmd+shift+r or cmd+click refresh), and see if the issue persists?

**Comment by yuuks** [comment]
ah yeah that seems to have been the issue. Looks all good now! 👍

**Review by yuuks — APPROVED** [review]

**Pull request merged**